### PR TITLE
Remove inital CN0 overshoots

### PR DESCRIPTION
**Pull request by Paakkit**
This update initializes the CN0 filter with average estimate of nsr.

### CODE REVIEW

File: src/track.c [hunk 1]
```diff
@@ -683,6 +683,8 @@ void cn0_est_init(cn0_est_state_t *s, float bw, float cn0_0,
   s->nsr_prev = s->nsr;
   s->xn = s->nsr;
   s->xn_prev = s->nsr;
+  s->avg_nsr = 0.f;
```
Comment (valeri-atamaniouk) [inline src/track.c]: Please reuse other fields. Make union as necessary.

File: src/track.c [hunk 2]
```diff
@@ -683,6 +683,8 @@ void cn0_est_init(cn0_est_state_t *s, float bw, float cn0_0,
...
+  s->init_count = 0;
```
Comment (valeri-atamaniouk) [inline src/track.c]: Same here

File: src/track.c [hunk 3]
```diff
@@ -775,8 +777,33 @@ float cn0_est(cn0_est_state_t *s, const cn0_est_params_t *p, float I, float Q)
     /* This is the first iteration, just update the prev state. */
     s->I_prev_abs = fabsf(I);
     s->Q_prev_abs = fabsf(Q);
+  } else if (s->init_count < 20) {
+	  /* This is the 20 first iterations, computing average estimate of nsr. */
```
Comment (valeri-atamaniouk) [inline src/track.c]: No tabs. 2 space indent

File: src/track.c [hunk 4]
```diff
@@ -775,8 +777,33 @@ float cn0_est(cn0_est_state_t *s, const cn0_est_params_t *p, float I, float Q)
...
+	  P_n = P_n*P_n;
+
+	  P_s = 0.5f*(I*I + s->I_prev_abs*s->I_prev_abs);
+
+	  s->I_prev_abs = fabsf(I);
+	  s->Q_prev_abs = fabsf(Q);
+
+	  /* This is to avoid division by zero. */
+	  if (P_s == 0.0) {
+	    return p->log_bw - 10.f*log10f(s->nsr);
```
Comment (adel-mamin) [inline src/track.c]: Does the code require a special handling, If s->nsr is 0?

**Pull request merged**